src/Native/System.Native/pal_io.cpp

```diff
@@ -383,13 +383,40 @@ extern "C" int32_t SystemNative_Pipe(int32_t pipeFds[2], int32_t flags)
 
     int32_t result;
 #if HAVE_PIPE2
+    // If pipe2 is available, use it.  This will handle O_CLOEXEC if it was set.
     while (CheckInterrupted(result = pipe2(pipeFds, flags)));
 #else
-    while (CheckInterrupted(result = pipe(pipeFds)));         // CLOEXEC intentionally ignored on platforms without pipe2.
+    // Otherwise, use pipe.
+    while (CheckInterrupted(result = pipe(pipeFds)));
+    
+    // Then, if O_CLOEXEC was specified, use fcntl to configure the file descriptors appropriately.
+    if ((flags & O_CLOEXEC) != 0 && result == 0)
+    {
+        while (CheckInterrupted(result = fcntl(pipeFds[0], F_SETFD, FD_CLOEXEC)));
```

Comment:
I went back and forth on that.  Ended up here since calling that would extend the int32_t to an intptr_t only to bring it back to an int32_t again.  But I'm fine either way... you want it changed?